pkg/server/cleaner.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/gocardless/draupnir/pkg/exec"
+	"github.com/gocardless/draupnir/pkg/models"
+	"github.com/gocardless/draupnir/pkg/store"
+	"github.com/prometheus/common/log"
+	google "google.golang.org/api/oauth2/v1"
+)
+
+type InstanceCleaner struct {
+	logger        log.Logger
+	instanceStore store.InstanceStore
+	executor      exec.Executor
+	service       googleService
+}
+
+type googleService interface {
+	Tokeninfo() *google.TokeninfoCall
+}
+
+func NewInstanceCleaner(logger log.Logger, instanceStore store.InstanceStore, executor exec.Executor, service googleService) *InstanceCleaner {
+	return &InstanceCleaner{
+		logger:        logger,
+		instanceStore: instanceStore,
+		executor:      executor,
+		service:       service,
+	}
+}
+
+func (ic *InstanceCleaner) Start(cctx context.Context, interval time.Duration) error {
+	for {
+		select {
+		case <-time.After(interval):
+			ic.logger.Info("cleaning old instances with invalid tokens")
+			instances, err := ic.instanceStore.List()
+			if err != nil {
+				ic.logger.Info("skipping cleaning old instances - could not get instances list")
+			} else {
+				for _, instance := range instances {
+					if instance.AccessToken != "" {
+						valid, err := ic.validateAccessToken(instance.AccessToken)
+						if err != nil {
+							ic.logger.With("instance", instance.ID).Error("failed to validate token: %s", err.Error())
+						} else if !valid {
+							ic.logger.With("instance", instance.ID).Info("token for instance invalid - destroying instance")
+							err = ic.destroyInstance(instance)
+							if err != nil {
+								ic.logger.With("instance", instance.ID).Error("failed to destroy instance: %s", err.Error())
```

Comment:
The format string isn't being used:
`msg="failed to destroy instance: %sexit status 1"`